implementations/rust/kex/src/xx.rs

```diff
@@ -106,31 +111,43 @@ impl KeyExchange for SymmetricState {
         &mut self,
         secret_handle: SecretKeyContext,
         public_key: PublicKey,
-    ) -> Result<Vec<u8>, VaultFailError> {
+    ) -> Result<Vec<SecretKeyContext>, VaultFailError> {
         let mut vault = self.vault.lock().unwrap();
```

Comment:
could we handle this w/o the `unwrap()`?

Reply:
@mikelodder7

Reply:
@mikelodder7  -- can you double-check that this is an acceptable `unwrap` here?

Reply:
The unwrap is okay on the mutex.